src/nvim/ex_cmds.c

```diff
@@ -3112,6 +3118,9 @@ void do_sub(exarg_T *eap)
   int start_nsubs;
   int save_ma = 0;
 
+  sub_done = 0;
```

Comment:
No, it can be `true` here when changing the `ics` setting from `""` to something else. We could reset it somewhere else (say, [here](https://github.com/neovim/neovim/pull/5226/files#diff-b9d70f1e57978ae9fbf83f73315b0c0fR6188)), or make it a "nontritivial `ics` setting only" variable, but the code would get more complicated. I'd vote for keeping it the way it is.